src/components/common/Dialogs/CreateDomainDialog/CreateDomainDialogForm.tsx

```diff
@@ -0,0 +1,123 @@
+import React from 'react';
+import { ColonyRole, Id } from '@colony/colony-js';
+import { defineMessages } from 'react-intl';
+import { useFormContext } from 'react-hook-form';
+
+import {
+  ActionDialogProps,
+  DialogControls,
+  DialogHeading,
+  DialogSection,
+} from '~shared/Dialog';
+import { HookFormInput as Input, Annotations } from '~shared/Fields';
+import NoPermissionMessage from '~shared/NoPermissionMessage';
+import CannotCreateMotionMessage from '~shared/CannotCreateMotionMessage';
+import PermissionRequiredInfo from '~shared/PermissionRequiredInfo';
+import DomainNameAndColorInputGroup from '~shared/DomainNameAndColorInputGroup';
+// import NotEnoughReputation from '~dashboard/NotEnoughReputation';
+import { noMotionsVotingReputationVersion } from '~utils/colonyMotions';
+
+import { useDialogActionPermissions, useEnabledExtensions } from '~hooks';
+
+const displayName = 'common.CreateDomainDialog.CreateDomainDialogForm';
+
+const MSG = defineMessages({
+  titleCreate: {
+    id: `${displayName}.titleCreate`,
+    defaultMessage: 'Create a new team',
+  },
+  purpose: {
+    id: `${displayName}.name`,
+    defaultMessage: 'What is the purpose of this team?',
+  },
+  annotation: {
+    id: `${displayName}.annotation`,
+    defaultMessage: 'Explain why you’re creating this team (optional)',
+  },
+});
+
+const CreateDomainDialogForm = ({ back, colony }: ActionDialogProps) => {
+  const {
+    formState: { isValid, isSubmitting },
+    getValues,
+  } = useFormContext();
+  const values = getValues();
+  const { votingReputationVersion, isVotingReputationEnabled } =
+    useEnabledExtensions(colony);
+
+  const requiredRoles: ColonyRole[] = [ColonyRole.Architecture];
+  const [userHasPermission, onlyForceAction] = useDialogActionPermissions(
+    colony,
+    isVotingReputationEnabled,
+    requiredRoles,
+    [Id.RootDomain],
+  );
+
+  const inputDisabled = !userHasPermission || onlyForceAction || isSubmitting;
+
+  const cannotCreateMotion =
```

Comment:
Could the logic above the view here be wrapped up into a hook that only exposes the data needed by the view? (eg `useCreateDomainDialogForm`)